src/%IPM/Main.cls

```diff
@@ -2383,11 +2408,208 @@ ClassMethod UpdateLanguageExtensions(pVerbose As %Boolean = 0, pTestOnly As %Boo
+	Quit
+ }
 $$$ENDTAG
 #; Need an extra line down here to avoid the end bracket being on the same line as $$$ENDTAG - a comment is just fine.
 }
 
+ClassMethod EnableIPM(ByRef pCommandInfo)
+{
+	Write !, "Version of IPM in current namespace: "
+	Do ..GetListModules("zpm",.list)
+	If $Data(list("zpm")) {
+			Write !,($namespace)_"> "_$$$FormattedLine($$$Green,"zpm ")_$ListGet(list("zpm"),1)
+	}
+	Kill list
+
+	Set quiet = $$$HasModifier(pCommandInfo,"quiet")
+	Set globally = $$$HasModifier(pCommandInfo, "globally")
+	Set localOnly = $$$HasModifier(pCommandInfo, "localOnly")
+	Set version = $$$GetModifier(pCommandInfo, "version")
+	Set namespaces = $$$GetModifier(pCommandInfo, "namespaces")
+	Set useLocal = 1			// var to store the final decision of whether to use local manifest or get from server
+	Set targetVersion = "" 		// var to store the final version of IPM to be installed
+	Kill targetNamespaces 		// multi-dim array to store the final namespaces that need to install IPM
+
+	// basic sanity checks
+	If (quiet && (namespaces = "") && 'globally) {
+		$$$ThrowOnError($$$ERROR($$$GeneralError,"Cannot install IPM in non-prompt mode without specifying namespaces to install."))
+	}
+	If ((namespaces '= "") && globally) {
+		$$$ThrowOnError($$$ERROR($$$GeneralError,"Cannot specify namespaces and global installation flag at the same time."))
+	}
+
+
+	// Get list of local IPM artifacts under /<instance install dir>/lib/ipm/
+	Set XMLDir = $system.Util.InstallDirectory()
+	For subDirName = "lib", "ipm" {
+		Set XMLDir = ##class(%File).SubDirectoryName(XMLDir, subDirName)
+	}
+  	Set statement = ##class(%SQL.Statement).%New()
+  	Set sc = statement.%PrepareClassQuery("%File", "FileSet")
+  	If $$$ISERR(sc) {
+		Do $SYSTEM.Status.DisplayError(sc)
+		Quit
+	}
+	// Valid IPM installation manifest should be of format: ipm-0.0.1.xml
+	Set resultSet = statement.%Execute(XMLDir, "ipm-*.xml")
+	If ((resultSet.%SQLCODE '= 0) && '(resultSet.%Message [ "does not exist")) {
+		Write !, resultSet.%Message 
+		Quit
+	}
+	Kill ipmLocalArray			// multi-dim array to store local ipm manifests; ipmLocalArray(<version>)=<fileName>
+	Kill menuList				// menu to let user choose which version of local ipm to install (if not in quiet mode and no version is specified)
+	While resultSet.%Next(.sc) {
+		$$$ThrowOnError(sc)
+		Set fileName = resultSet.%Get("Name")
+		If (resultSet.%Get("Type") = "F") {
+			// get the exact version string from filename
+			Set exactIPMVer = $Piece($Piece(fileName, "ipm-", 2, *), ".xml")
+			Set ipmLocalArray(exactIPMVer) = fileName
+			Set menuList($Increment(menuList)) = exactIPMVer_" (local version)"
+		}
+	}
+	$$$ThrowOnError(sc)
+
+	If ('$DATA(menuList) && localOnly) {
+		$$$ThrowOnError($$$ERROR($$$GeneralError,"No ipm-<version>.xml installer file is found locally in directory: "__XMLDir))
+	}
+
+	If (version '= "") {
+		Set targetVersion = version
+		If '$DATA(ipmLocalArray(version)) {
```

Comment:
Hmm can you create a Jira for deciding whether we want to support always obtaining a version from the registry? Right now if a local version is available, we are always using that but may want to force getting from registry.
In future, once packages have hashes (not sure if they do), we could do a hash check against the registry and local versions to confirm they are the same.

Reply:
Sounds good, will create a jira for it and describe the current settings we have (in which case do local, in which case do remote, etc.)